From 7b96c6e15ada9f07e72d56f4651e7bca1148b531 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Thu, 19 Dec 2024 14:38:46 +0100
Subject: [PATCH] build without -Wsign-compare

---
 nvidia-modules-common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvidia-modules-common.mk b/nvidia-modules-common.mk
index b32bca76..0f1b5166 100644
--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -38,7 +38,7 @@ obj-m := $(MODULE_NAME).o
 #
 
 EXTRA_CFLAGS += -I$(src)
-EXTRA_CFLAGS += -Wall -MD $(DEFINES) $(INCLUDES) -Wsign-compare -Wno-cast-qual -Wno-error
+EXTRA_CFLAGS += -Wall -MD $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error
 
 #
 # Output directory for the build; default to the source directory if not set.
-- 
2.39.5

