From e746e9bf9db46f06d5f48819a77c2c642bb27c67 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 27 Jan 2026 17:41:05 +0100
Subject: [PATCH] backport drm_print.h changes from 570.211.01

---
 conftest.sh | 1 +
 nv-drm.c    | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index 6154c24c..3e359786 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -149,6 +149,7 @@ test_headers() {
     #
     FILES="asm/system.h"
     FILES="$FILES drm/drmP.h"
+    FILES="$FILES drm/drm_print.h"
     FILES="$FILES drm/drm_gem.h"
     FILES="$FILES drm/drm_drv.h"
     FILES="$FILES drm/drm_prime.h"
diff --git a/nv-drm.c b/nv-drm.c
index f6fb4583..5f93a277 100644
--- a/nv-drm.c
+++ b/nv-drm.c
@@ -98,6 +98,10 @@
 #include <linux/file.h>
 #endif
 
+#if defined(NV_DRM_DRM_PRINT_H_PRESENT)
+#include <drm/drm_print.h>
+#endif
+
 #if defined(NV_DRM_DRM_DRV_H_PRESENT)
 #include <drm/drm_drv.h>
 #endif
-- 
2.39.5

