From bebc9af5f0f0a587210d5ceff891ed4a3e631525 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Thu, 19 Dec 2024 13:42:55 +0100
Subject: [PATCH] backport cmd_symlink changes from 550.142

---
 nvidia-modules-common.mk | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/nvidia-modules-common.mk b/nvidia-modules-common.mk
index 0f1b5166..6a8b15d5 100644
--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -138,6 +138,19 @@ MOD_SIGN_CMD ?= $(KERNEL_SOURCES)/scripts/sign-file
 # to satisfy KBUILD requirements and to support NVIDIA specifics.
 #
 
+#
+# Command to create a symbolic link, explicitly resolving the symlink target
+# to an absolute path to abstract away the difference between Linux < 6.13,
+# where the CWD is the Linux kernel source tree for Kbuild extmod builds, and
+# Linux >= 6.13, where the CWD is the external module source tree.
+#
+# This is used to create the nv*-kernel.o -> nv*-kernel.o_binary symlinks for
+# kernel modules which use precompiled binary object files.
+#
+
+quiet_cmd_symlink = SYMLINK $@
+ cmd_symlink = ln -sf $(abspath $<) $@
+
 $(obj)/$(CORE_OBJS):
 	@cp $(src)/$(CORE_OBJS) $(obj)/$(CORE_OBJS)
 
-- 
2.39.5

