From 149aadf9282b21426555c58adc05775897cd5122 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 10 Dec 2024 20:08:50 +0100
Subject: [PATCH] fix more uvm warnings

---
 nvidia-uvm/uvm_gpu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nvidia-uvm/uvm_gpu.c b/nvidia-uvm/uvm_gpu.c
index 62132ab5..15cac0b0 100644
--- a/nvidia-uvm/uvm_gpu.c
+++ b/nvidia-uvm/uvm_gpu.c
@@ -1762,6 +1762,7 @@ uvm_gpu_t *uvm_gpu_get_by_uuid(const NvProcessorUuid *gpu_uuid)
     return uvm_gpu_get_by_uuid_locked(gpu_uuid);
 }
 
+static
 uvm_gpu_t *uvm_gpu_get_by_uuid_and_swizz_id_locked(const NvProcessorUuid *gpu_uuid, NvU32 swizz_id)
 {
     uvm_parent_gpu_t *parent_gpu = uvm_parent_gpu_get_by_uuid(gpu_uuid);
-- 
2.39.5

