From: Andreas Beckmann <debian@abeckmann.de>
Subject: Fix build of 173xx with 2.6.33 and newer kernels

Index: b/nv-linux.h
===================================================================
--- a/nv-linux.h	2010-06-10 15:24:01.824037317 +0000
+++ b/nv-linux.h	2010-06-10 15:24:25.120037815 +0000
@@ -15,11 +15,13 @@
 #include "nv.h"
 #include "conftest.h"
 
+#ifndef AUTOCONF_INCLUDED
 #if defined(NV_GENERATED_AUTOCONF_H_PRESENT)
 #include <generated/autoconf.h>
 #else
 #include <linux/autoconf.h>
 #endif
+#endif
 
 #ifndef LINUX_VERSION_CODE
 #include <linux/version.h>
