Description: fix module build for Linux 3.11
Author: Andreas Beckmann <anbe@debian.org>

backported from 173.14.38

index 6c1ae8e..9948266 100644
--- a/os-interface.c
+++ b/os-interface.c
@@ -292,7 +292,15 @@ NvU64 NV_API_CALL os_get_page_mask(void)
 
 NvU64 NV_API_CALL os_get_system_memory_size(void)
 {
-    return ((NvU64) num_physpages * PAGE_SIZE) / RM_PAGE_SIZE;
+    NvU64 num_pages;
+
+#if defined(NV_GET_NUM_PHYSPAGES_PRESENT)
+    num_pages = get_num_physpages();
+#else
+    num_pages = num_physpages;
+#endif
+
+    return ((num_pages * PAGE_SIZE) / RM_PAGE_SIZE);
 }
 
 //
