From 6499352d257956c9458fe5076f1a42d9e44a3ddb Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 10 Dec 2024 20:08:50 +0100
Subject: [PATCH] fix more uvm warnings

---
 nvidia-uvm/uvm_gpu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nvidia-uvm/uvm_gpu.c b/nvidia-uvm/uvm_gpu.c
index 9070f444..f7874a9e 100644
--- a/nvidia-uvm/uvm_gpu.c
+++ b/nvidia-uvm/uvm_gpu.c
@@ -1881,6 +1881,7 @@ uvm_gpu_t *uvm_gpu_get_by_uuid(const NvProcessorUuid *gpu_uuid)
     return uvm_gpu_get_by_uuid_locked(gpu_uuid);
 }
 
+static
 uvm_gpu_t *uvm_gpu_get_by_parent_and_swizz_id_locked(uvm_parent_gpu_t *parent_gpu, NvU32 swizz_id)
 {
     uvm_gpu_t *gpu;
-- 
2.39.5

